### PR TITLE
Parsing assembly function definitions and calls.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: test/libsolidity/InlineAssembly.cpp [hunk 1]
```diff
@@ -148,6 +148,26 @@ BOOST_AUTO_TEST_CASE(blocks)
...
 }
 
+BOOST_AUTO_TEST_CASE(function_definitions)
+{
+	BOOST_CHECK(successParse("{ function f() { } function g(a) -> (x) { } }"));
+}
+
+BOOST_AUTO_TEST_CASE(function_definitions_multiple_args)
+{
+	BOOST_CHECK(successParse("{ function f(a, d) { } function g(a, d) -> (x, y) { } }"));
```
Comment (axic) [inline test/libsolidity/InlineAssembly.cpp:169]: Should have a failing test also for conflicting identifiers (since the code checks for that).

Comment (chriseth) [reply test/libsolidity/InlineAssembly.cpp:169]: Are you talking about function names? This is just the parser for now, it does not check for that.

**Review by yann300 — APPROVED** [review]

File: test/libsolidity/InlineAssembly.cpp (hunk 1 above)
Comment (axic) [reply test/libsolidity/InlineAssembly.cpp:169]: Conflicting identifiers, it does for function name, parameters and return values:
```
funDef.name = expectAsmIdentifier();
funDef.arguments.push_back(expectAsmIdentifier());
funDef.returns.push_back(expectAsmIdentifier());
```

Comment (chriseth) [reply test/libsolidity/InlineAssembly.cpp:169]: Ah you mean opcodes for identifiers and not things like `function f(a, a)`?

Comment (axic) [reply test/libsolidity/InlineAssembly.cpp:169]: Yep, those are already a given. We should of course check for duplicates and have tests for that, but that's not implemented yet.

**Pull request merged**